gapis/trace/android/adreno/profiling_data.go

```diff
@@ -90,6 +90,45 @@ func extractTraceHandles(ctx context.Context, replayHandles *[]int64, replayHand
 	}
 }
 
+func fixContextIds(contextIDs []int64) []int64 {
```

Comment:
Could you add a function comment on what this is intended to do?

For example something like this:
// fixContextIds applies heuristics to fill in missing context ids.
// This is to work around a bug in certain devices (QC, see b/...) reporting a device id of 0.
// If only one non-zero device id is found, all zero device ids are replaced by that id.
// Otherwise the contextIds are left unchanged.

Reply:
I have merged the two comment about comments and answering them here:

I generally do not favor explaining the code. It either leads to outdated comment, or extra maintenance of the comment. I think code itself should be, and in this case I believe it is, explaining itself.

Here however there is an extra bit of it that makes people aware the bug and the logic to workaround it. So I added the information about why and how we do the workaround.

Reply:
And for being a bit more honest. I removed the "temporary" from the comment. Realistically speaking, we will never remove this if this is not a perfetto but a QC bug because there will always be devices with old drivers.

Reply:
I have added more comment and I think it should be fine with already existing comments on other lines.

Reply:
The comment contains all the information. I'm very used to the typical function comment style and think there is value in consistency/making code easier to read that way. I.e. the suggestion is to replace the comment inside the function.

It's a bit of a style question, as I'm new to this and don't really understand if there is consistency or consensus on this, feel free to leave as-is.